validate syntactic section

connor-a-francis opened the pull request:
This pull request should implement the following validations

- LHS Non-terminal names must start with a lower-case letter, and may contain upper or lower case letters, numbers, and underscore.
- The resolved names for all LHS rules must be unique.
- User supplied names on for LHS symbols must start with a capital letter, and may contain upper or lower case letters, numbers, and underscore.

StoneyJackson commented on src/plcc/load_spec/validate_spec/validate_syntactic_spec/errors.py:
```diff
@@ -0,0 +1,30 @@
+@dataclass
+class ValidationError:
+    def __init__(self, rule):
+        self.rule = rule
```
This constructor is never called. Derived classes are directly running `self.rule = rule`. Instead, the should be running super().__init__(rule).

StoneyJackson commented on src/plcc/load_spec/validate_spec/validate_syntactic_spec/validate_rhs.py:
```diff
@@ -0,0 +1,18 @@
+)
+
+
+def validate_rhs(syntacticSpec: SyntacticSpec):
```
No test is calling validate_rhs. If it's not needed, please remove. Or figure out where it should be called, and/or figure out the condition it should be called and make sure there is a test covering that situation.

StoneyJackson commented on src/plcc/load_spec/validate_spec/validate_syntactic_spec/validate_rhs.py:
```diff
@@ -0,0 +1,18 @@
+class SyntacticRhsValidator:
+    spec: SyntacticSpec
+
+    def __init__(self, syntacticSpec: SyntacticSpec):
```
This constructor is not called under any test.

StoneyJackson commented on src/plcc/load_spec/validate_spec/validate_syntactic_spec/validate_rhs.py:
```diff
@@ -0,0 +1,18 @@
+        self.spec = syntacticSpec
+        self.errorList = []
+
+    def validate(self):
```
This method is never called by any test.

StoneyJackson commented on src/plcc/load_spec/validate_spec/validate_syntactic_spec/validate_syntactic_spec.py:
```diff
@@ -0,0 +1,35 @@
+            self.errorList = lhs_error_list
+        self.nonTerminals = non_terminal_set
+
+    def _validateRhs(self):
```
This is never called.

StoneyJackson reviewed (changes requested):
Please review the coverage report and take a look at the lines in the files that you have modified. The report is telling you that those lines of code (or a branch in your code) was not exercised by any test. That means either the code is not necessary and should be removed, or there is no test covering that code's functionality.

connor-a-francis commented:
Hello! I was laying down plumbing so folks could work on the RHS validator without colliding. I can do that in a separate MR if you would like. 

@StoneyJackson

StoneyJackson commented:
@connor-a-francis Ah, ok. Let's leave them in.

OK, squash it and let's get it merged!

connor-a-francis committed:
feat: validate lhs of syntactic section

Validates the LHS of a given Syntactic Spec, as well as laying
foundation for RHS to be validated.
---

* Closes #22

connor-a-francis commented:
Squashed @StoneyJackson!
